splice_machine/src/main/java/com/splicemachine/derby/stream/control/ControlDataSetWriter.java

```diff
@@ -64,26 +76,59 @@ public ControlDataSetWriter(ControlDataSet<ExecRow> dataSet, AbstractPipelineWri
     @Override
     public DataSet<ExecRow> write() throws StandardException{
         SpliceOperation operation=operationContext.getOperation();
-
+        LanguageConnectionContext lcc =
+            operationContext.getActivation().getLanguageConnectionContext();
         try{
-            boolean inMemoryTxn = !operation.isOlapServer();
+            boolean inMemoryTxn = !operation.isOlapServer() && !SpliceClient.isClient();
```

Comment:
Why is `!SpliceClient.isClient` needed here? (And why wasn't it needed before this PR?)